File: 0509-fibonacci-number/0509-fibonacci-number.py
class Solution(object):
    def fib(self, n):
        fib=[0,1]
        if n==0:
            return n
        for i in range(2,n+1):
            fib.append(fib[i-1] + fib[i-2])
        return fib[-1]
        

File: 0509-fibonacci-number/0509-fibonacci-number.java
class Solution 
{
    public int fib(int n) 
    {
        if (n <= 1) 
            return n;
        int a = 0, b = 1;
        for (int i = 2; i <= n; i++) 
        {
            int temp = a + b;
            a = b;
            b = temp;
        }
        return b;
    }
}



// class Solution 
// {
//     int[] arr;
//     public int fib(int n) 
//     {
//         arr = new int[n+1];
//         return solve(n);
//     }
//     public int solve(int n)
//     {
//         if(n<=1)
//             return n;
//         return solve(n-1)+solve(n-2);
//     }
// }